### PR TITLE
Pil 416 grs flow

### PR DESCRIPTION
**Pull request by olakunlekuye2**

### CODE REVIEW

File: app/controllers/subscription/CaptureContactAddressController.scala [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+      .get(SubscriptionPage)
+      .fold(false) { data =>
+        data.groupDetailStatus.toString == "Completed"
+      }
+
+  private def isNfmNotRegisteredUK(request: DataRequest[AnyContent]): Boolean =
+    request.userAnswers
+      .get(NominatedFilingMemberPage)
+      .fold(false) { data =>
+        data.isNfmRegisteredInUK.fold(false)(regInUk => regInUk == NfmRegisteredInUkConfirmation.No)
```
Comment (AKKUM) [inline app/controllers/subscription/CaptureContactAddressController.scala:262]: Looks like the branch is away from main branch. NfmRegisteredInUkConfirmation is not there anymore. Lets have a look after rebasing.

File: app/controllers/subscription/CaptureContactAddressController.scala [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+      }
+
+  private def isUpeRegisteredUK(request: DataRequest[AnyContent]): Boolean =
+    request.userAnswers
+      .get(RegistrationPage)
+      .fold(false) { data =>
+        data.isUPERegisteredInUK == UPERegisteredInUKConfirmation.No
```
Comment (AKKUM) [inline app/controllers/subscription/CaptureContactAddressController.scala:269]: Same here.

File: app/controllers/subscription/CaptureContactAddressController.scala (hunk 1 above)
Comment (olakunlekuye2) [reply app/controllers/subscription/CaptureContactAddressController.scala:262]: Oh okay I do that

File: app/controllers/subscription/CaptureContactAddressController.scala (hunk 2 above)
Comment (olakunlekuye2) [reply app/controllers/subscription/CaptureContactAddressController.scala:269]: Okay

File: app/controllers/subscription/CaptureContactAddressController.scala [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+      }
+
+  private def getName(request: DataRequest[AnyContent]): String = {
+    val registration = request.userAnswers.get(NominatedFilingMemberPage)
+    registration.fold("")(regData => regData.withoutIdRegData.fold("")(withoutId => withoutId.fmContactName.fold("")(name => name)))
+  }
+
+  private def getEmail(request: DataRequest[AnyContent]): String = {
```
Comment (nitin-srivastav) [inline app/controllers/subscription/CaptureContactAddressController.scala:276]: This method is not required here

File: app/controllers/subscription/CaptureContactAddressController.scala [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+    val registration = request.userAnswers.get(NominatedFilingMemberPage)
+    registration.fold("")(regData => regData.withoutIdRegData.fold("")(withoutId => withoutId.fmEmailAddress.fold("")(email => email)))
+  }
+
+  private def getPhoneNumber(request: DataRequest[AnyContent]): String = {
```
Comment (nitin-srivastav) [inline app/controllers/subscription/CaptureContactAddressController.scala:281]: This method is not required here

File: app/controllers/subscription/CaptureContactAddressController.scala [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+    val registration = request.userAnswers.get(NominatedFilingMemberPage)
+    registration.fold("")(regData => regData.withoutIdRegData.fold("")(withoutId => withoutId.telephoneNumber.fold("")(tel => tel)))
+  }
+
+  private def getUpeName(request: DataRequest[AnyContent]): String = {
```
Comment (nitin-srivastav) [inline app/controllers/subscription/CaptureContactAddressController.scala:286]: This method is not required here

File: app/controllers/subscription/CaptureContactAddressController.scala [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+    val registration = request.userAnswers.get(RegistrationPage)
+    registration.fold("")(regData => regData.withoutIdRegData.fold("")(withoutId => withoutId.upeContactName.fold("")(name => name)))
+  }
+
+  private def getUpeEmail(request: DataRequest[AnyContent]): String = {
```
Comment (nitin-srivastav) [inline app/controllers/subscription/CaptureContactAddressController.scala:291]: This method is not required here

File: app/controllers/subscription/CaptureContactAddressController.scala [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+    val registration = request.userAnswers.get(RegistrationPage)
+    registration.fold("")(regData => regData.withoutIdRegData.fold("")(withoutId => withoutId.emailAddress.fold("")(email => email)))
+  }
+
+  private def getUpePhoneNumber(request: DataRequest[AnyContent]): String = {
```
Comment (nitin-srivastav) [inline app/controllers/subscription/CaptureContactAddressController.scala:296]: This method is not required here

**Comment by olakunlekuye2** [comment]
no longer required

**Pull request closed without merging**